datafusion/physical-expr/src/analysis.rs

```diff
@@ -335,7 +355,9 @@ mod tests {
                 df_schema.as_ref(),
             )
             .unwrap();
-            let actual = &analysis_result.boundaries[0].interval;
+            let Some(actual) = &analysis_result.boundaries[0].interval else {
+                panic!("Interval should be initialized for all columns");
```

Comment:
```suggestion
                panic!("The analysis result should contain non-empty intervals for all columns");
```
AFAICT this is not about initialization, you expect non-empty results.